src/mqt/bench/targets/gatesets/rigetti.py

```diff
@@ -0,0 +1,173 @@
+        q_0: ┤ RX(π)├
+             └─────┘
+
+    **Matrix representation:**
+
+    .. math::
+
+        RX(π) =
+            \begin{pmatrix}
+                0 & -i \\
+                -i & 0
+            \end{pmatrix}
+    """
+
+    def __init__(self, label: str | None = None) -> None:
+        """Create RX(π) gate."""
+        super().__init__("fixed_angle_rx_pi", 1, [np.pi], label=label)
+
+    def _define(self) -> None:
+        """Define RX(π) gate using standard RX."""
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.rx(np.pi, 0)
+        self.definition = qc
+
+
+class RXPiOver2Gate(Gate):  # type: ignore[misc]
+    r"""Single-qubit RX(π/2) gate.
+
+    **Circuit symbol:**
+    .. parsed-literal::
+             ┌──────────┐
+        q_0: ┤ RX(π/2)  ├
+             └──────────┘
+
+    **Matrix representation:**
+
+    .. math::
+
+        RX(π/2) =
+            \begin{pmatrix}
+                \cos(π/4) & -i \sin(π/4) \\
+                -i \sin(π/4) & \cos(π/4)
+            \end{pmatrix}
+    """
+
+    def __init__(self, label: str | None = None) -> None:
+        """Create RX(π/2) gate."""
+        super().__init__("fixed_angle_rx_pi_by_2", 1, [np.pi / 2], label=label)
+
+    def _define(self) -> None:
+        """Define RX(π/2) gate using standard RX."""
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.rx(np.pi / 2, 0)
+        self.definition = qc
+
+
+class RXMinusPiOver2Gate(Gate):  # type: ignore[misc]
+    r"""Single-qubit RX(-π/2) gate.
+
+    **Circuit symbol:**
+    .. parsed-literal::
+             ┌────────────┐
+        q_0: ┤ RX(-π/2)   ├
+             └────────────┘
+
+    **Matrix representation:**
+
+    .. math::
+
+        RX(-π/2) =
+            \begin{pmatrix}
+                \cos(π/4) & i \sin(π/4) \\
+                i \sin(π/4) & \cos(π/4)
+            \end{pmatrix}
+    """
+
+    def __init__(self, label: str | None = None) -> None:
+        """Create RX(-π/2) gate."""
+        super().__init__("fixed_angle_rx_pi_by_minus_2", 1, [-np.pi / 2], label=label)
+
+    def _define(self) -> None:
+        """Define RX(-π/2) gate using standard RX."""
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.rx(-np.pi / 2, 0)
+        self.definition = qc
+
+
+def u_gate_equivalence() -> None:
+    """Add U(θ, φ, λ) gate equivalence to the SessionEquivalenceLibrary using RZ and RX gates."""
+    theta = Parameter("θ")
+    phi = Parameter("φ")
+    lam = Parameter("λ")
+
+    qr = QuantumRegister(1, "q")
+    qc = QuantumCircuit(qr)
+
+    # Decomposition: U(θ, φ, λ) = RZ(φ) RX(-π/2) RZ(θ) RX(π/2) RZ(λ)
+    qc.append(RZGate(phi), [qr[0]])
+    qc.append(RXMinusPiOver2Gate(), [qr[0]])
+    qc.append(RZGate(theta), [qr[0]])
+    qc.append(RXPiOver2Gate(), [qr[0]])
+    qc.append(RZGate(lam), [qr[0]])
+
+    SessionEquivalenceLibrary.add_equivalence(UGate(theta, phi, lam), qc)
+
+
+def cx_gate_equivalence() -> None:
```

Comment:
Is it really true that a single CX gate takes two iSWAP gates? Seems counterintuitive to me.
Furthermore, the iSWAP gate is a Qiskit standard gate, so this equivalence should not really be needed.

Reply:
I thinks so, also considering this reference implementation (https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.library.iSwapGate). Although it is the other way round there. Anyways, I removed this equivalence because it is not necessary.